React/Modules/RCTStatusBarManager.m

```diff
@@ -72,35 +70,29 @@ - (dispatch_queue_t)methodQueue
   return dispatch_get_main_queue();
 }
 
-- (void)emitEvent:(NSString *)eventName forNotification:(NSNotification *)notification
+static NSDictionary *RCTExportedFrame()
 {
-  CGRect frame = [notification.userInfo[UIApplicationStatusBarFrameUserInfoKey] CGRectValue];
-  NSDictionary *event = @{
-    @"frame": @{
-      @"x": @(frame.origin.x),
-      @"y": @(frame.origin.y),
-      @"width": @(frame.size.width),
-      @"height": @(frame.size.height),
-    },
-  };
-  [self sendEventWithName:eventName body:event];
+  // The In-Call Status Bar pushes the root view down instead of increasing the size of the StatusBar area
+  // that needs to be accounted for in JS. We can fix that by subtracting the y value of the root view
+  // from the StatusBar's height.
+  CGRect rootFrame = RCTPresentedViewController().view.frame;
+  CGRect statusBarFrame = RCTSharedApplication().statusBarFrame;
+  return @{
+           @"y": @(rootFrame.origin.y),
+           @"height": @(statusBarFrame.size.height - rootFrame.origin.y),
+           };
 }
 
-- (void)applicationDidChangeStatusBarFrame:(NSNotification *)notification
+- (void)applicationDidChangeStatusBarFrame
 {
-  [self emitEvent:@"statusBarFrameDidChange" forNotification:notification];
-}
-
-- (void)applicationWillChangeStatusBarFrame:(NSNotification *)notification
-{
-  [self emitEvent:@"statusBarFrameWillChange" forNotification:notification];
+  dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
Why do we need this?

Reply:
I had issues with iOS deciding to return `0` instead of a proper value sometimes for the y/height, even `notification` didn't always behave right IIRC.

Reply:
I am affraid, that kind of hacks are unacceptable. This notification already fires on the main thread, and this `nextTick` looks very hacky and unreliable.